src/compass/utils/range_split_spectrum.py

```diff
@@ -0,0 +1,113 @@
+    half_bw = 0.5 * burst.range_bandwidth
+    low_bw = cfg_split_spectrum.low_band_bandwidth
+    high_bw = cfg_split_spectrum.high_band_bandwidth
+
+    low_freq_burst = burst.radar_center_frequency - half_bw
+    high_freq_burst = burst.radar_center_frequency + half_bw
+
+    low_band_freqs = np.array([low_freq_burst, low_freq_burst + low_bw])
+    high_band_freqs = np.array([high_freq_burst - high_bw, high_freq_burst])
+    low_center_freq = low_freq_burst + low_bw / 2
+    high_center_freq = high_freq_burst - high_bw / 2
+
+    # Initialize the split-spectrum parameter object. Note, constructor
+    # requires frequency (A/B) but this is not used anywhere below
+    rdr_grid = burst.as_isce3_radargrid()
+    split_spectrum_obj = splitspectrum.SplitSpectrum(
+        rg_sample_freq=burst.range_sampling_rate,
+        rg_bandwidth=burst.range_bandwidth,
+        center_frequency=burst.radar_center_frequency,
+        slant_range=rdr_grid.slant_range,
+        freq='A')
+
+    # Save the burst locally
+    burst_path = f'{scratch_path}/burst_temp.vrt'
+    burst.slc_to_vrt_file(burst_path)
+
+    # The output burst will
+    # contain 3 bands: Band #1: low-band image; Band #2 main-band image;
+    # Band #3: high-band image.
+    in_ds = gdal.Open(burst_path, gdal.GA_ReadOnly)
+    length = in_ds.RasterYSize
+    width = in_ds.RasterXSize
+    driver = gdal.GetDriverByName('ENVI')
+    out_ds = driver.Create(f'{scratch_path}/{burst.burst_id}_low_main_high',
+                           width, length, 3, gdal.GDT_CFloat32)
+
+    # Prepare necessary variables for block processing
+    lines_per_block = min(length, lines_per_block)
+    num_blocks = int(np.ceil(length / lines_per_block))
+
+    for block in range(num_blocks):
+        line_start = block * lines_per_block
+        if block == num_blocks - 1:
+            block_length = length - line_start
+        else:
+            block_length = lines_per_block
+
+        # Read a block of valid burst data
+        burst_data = in_ds.GetRasterBand(1).ReadAsArray(0, line_start,
+                                                        width, block_length)
+        # Get the low band sub-image and corresponding metadata
+        burst_low_data, _ = split_spectrum_obj.bandpass_shift_spectrum(
+            slc_raster=burst_data, low_frequency=low_band_freqs[0],
+            high_frequency=low_band_freqs[1],
+            new_center_frequency=low_center_freq,
+            fft_size=width, window_shape=1.0 - burst.range_window_coefficient,
+            window_function=window_type, resampling=False
+        )
+        # Get the high sub-image and corresponding metadata
+        burst_high_data, _ = split_spectrum_obj.bandpass_shift_spectrum(
+            slc_raster=burst_data, low_frequency=high_band_freqs[0],
+            high_frequency=high_band_freqs[1],
+            new_center_frequency=high_center_freq,
+            fft_size=width, window_shape=1.0 - burst.range_window_coefficient,
```

Comment:
Not related to this PR really. But let me ask @oberonia78 if the split spectrum can handle fft_size to be any number which equals to or larger than width? in particular does it accept "next power of 2" for efficiency of FFT computation? 
Does split spectrum module expect the user (here this PR) to find the next power of 2 and pass to fft_size here?

Reply:
@hfattahi Yes. fft_size could be any number if fft_size >= width. Since split_spectrum applies FFT using the given fft_size and computes the frequency range with the given information, fft_size could be any number. 
I thought "next power of 2" could be efficient in computation as you mentioned. But I did not find the big differences between "next power of 2" and image width. It might be because I tested out with UAVSAR images, which is usually smaller than satellite image.

Reply:
Thanks Jungkyo. Just to be sure, split spectrum expects the user to compute "next power of 2" and pass the corresponding fft size to the split spectrum? is that right?

Reply:
Nope. The split-spectrum implemented in the ISCE3 does not expect anything. For example, bandpass uses the image width as fft_size.

Reply:
Ok so it sounds like that splitspectrum does not change the fft_size. So it is user responsibility to introduce the fft_size with correct number. 
@vbrancat my suggestion is that you add a simple function somewhere in your utility to compute next power of 2:

here is one c++ [implementation](https://github.com/isce-framework/isce3/blob/develop/cxx/isce3/signal/Signal.icc#L19) which you can translate to python or we might have it somewhere. An overkill is [here](https://github.com/isce-framework/isce3/blob/develop/cxx/isce3/fft/FFTUtil.icc#L39). The idea is to find the number which is a power of 2 and is larger or equal to width

Once you have that function then compute fft_size from width and then pass the fft_size to split spectrum instead of passing width.